ui/apps/platform/src/Components/PatternFly/CheckboxSelect.tsx

```diff
@@ -48,23 +56,59 @@ function CheckboxSelect({
         }
     }
 
+    const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+            className="pf-v5-u-w-100"
+            id={toggleId}
+            ref={toggleRef}
+            onClick={onToggle}
+            isExpanded={isOpen}
+            icon={toggleIcon}
+            aria-label={ariaLabel}
+        >
+            <Flex
+                alignItems={{ default: 'alignItemsCenter' }}
+                spaceItems={{ default: 'spaceItemsSm' }}
+            >
+                <FlexItem>{placeholderText}</FlexItem>
+                {selections.length > 0 && <Badge isRead>{selections.length}</Badge>}
+            </Flex>
+        </MenuToggle>
+    );
+
+    // Automatically inject hasCheckbox and isSelected props
+    const enhancedChildren = React.Children.map(children, (child) => {
+        if (React.isValidElement(child) && child.type === SelectOption) {
```

Comment:
If the prop type to `ReactElement` above is changed, we can remove all of the type assertions here. Either way, it _might_ be beneficial to add a `'value' in child.props check here. Although `React.isValidElement()` ensures that this is safe, if that check is removed it is possible for a non-object `child.props` to make it through without type errors which would cause a crash during render.